op-service/client/http.go

```diff
@@ -23,16 +23,40 @@ type HTTP interface {
 
 type BasicHTTPClient struct {
 	endpoint string
-	log      log.Logger
-	client   *http.Client
+	header   http.Header // optional header to use in every request
+
+	log    log.Logger
+	client *http.Client
 }
 
-func NewBasicHTTPClient(endpoint string, log log.Logger) *BasicHTTPClient {
-	return &BasicHTTPClient{
+func NewBasicHTTPClient(endpoint string, log log.Logger, opts ...BasicHTTPClientOption) *BasicHTTPClient {
+	c := &BasicHTTPClient{
 		endpoint: endpoint,
 		log:      log,
 		client:   &http.Client{Timeout: DefaultTimeoutSeconds * time.Second},
 	}
+
+	for _, opt := range opts {
+		opt.Apply(c)
+	}
+
+	return c
+}
+
+type BasicHTTPClientOption interface {
+	Apply(c *BasicHTTPClient)
+}
+
+type BasicHTTPClientOptionFn func(*BasicHTTPClient)
+
+func (fn BasicHTTPClientOptionFn) Apply(c *BasicHTTPClient) {
+	(func(*BasicHTTPClient))(fn)(c)
```

Comment:
```suggestion
	fn(c)
```
No need to cast the type here.